third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -316,10 +317,13 @@ Value llLoad(RewriterBase &rewriter, Location loc, Value ptr, Type elemTy,
       return predicatedLoad;
     }
   };
+  std::string funcName = getLoadNameRaw(cm);
+  if (forceNoAliasAsyncLoads)
+    funcName += noAliasAsyncLoads;
 
-  auto funcName = mangleFunc(getLoadNameRaw(cm), funcType);
+  auto mangledName = mangleFunc(funcName, funcType);
```

Comment:
Not for this pull request but this name mangling stuff is historical and we want to improve it by defining some auxiliary ops in amdgpu dialect to avoid the string based match.. Can you create an internal tracking issue?

Reply:
https://github.com/ROCm/triton-internal/issues/1014